editor.planx.uk/src/pages/FlowEditor/lib/store/editor.ts

```diff
@@ -228,6 +229,28 @@ export const editorStore: StateCreator<
     send(ops);
   },
 
+  archiveFlow: (flowId, teamSlug) => {
+    const valid = get().canUserEditTeam(teamSlug);
+    if (!valid) {
+      alert(`You do not have permission to archive this flow from ${teamSlug}`);
+      return Promise.resolve();
+    }
```

Comment:
This validation is now happening multiple times - 
 - In the UI displaying the archive button
 - Here
 - `useTeamAuth` middleware
 - Hasura permissions
 
It's good to be safe but this is a bit much 😅 If permissions change we'll need to remember to keep all of these in sync.